### PR TITLE
Install merbridge cni

### PR DESCRIPTION
**Pull request by dddddai**
Part of #118

### CODE REVIEW

File: internal/cni-server/server.go [hunk 1]
```diff
@@ -43,7 +43,7 @@ type server struct {
 // the path this the unix path to listen.
 func NewServer(unixSockPath string, bpfMountPath string) Server {
 	if unixSockPath == "" {
-		unixSockPath = "/var/run/merbridge-cni.sock"
+		unixSockPath = "/host/var/run/merbridge-cni.sock"
```
Comment (kebe7jun) [inline internal/cni-server/server.go:46]: cni plugin is running on host, not in container.
so, it should be `/var/run/merbridge-cni.sock`.

File: pkg/cniplugin/kube.go [hunk 2]
```diff
@@ -49,7 +49,7 @@ func newKubeClient(conf plugin.Config) (*kubernetes.Clientset, error) {
 		return nil, err
 	}
 
-	log.Debugf("istio-cni set up kubernetes client with kubeconfig %s", kubeconfig)
+	log.Debugf("mb-cni set up kubernetes client with kubeconfig %s", kubeconfig)
```
Comment (kebe7jun) [inline pkg/cniplugin/kube.go:52]: use `merbridge-cni` instead?

File: Dockerfile [hunk 3]
```diff
@@ -26,6 +26,7 @@ RUN go mod download
 ADD . .
 
 RUN go build -ldflags "-s -w" -o ./dist/mbctl ./app/main.go
+RUN go build -ldflags "-s -w" -o ./dist/mb-cni ./app/cni/main.go
```
Comment (kebe7jun) [inline Dockerfile:29]: use `merbridge-cni` instead is better?

File: app/cmd/root.go [hunk 4]
```diff
@@ -35,11 +38,12 @@ var rootCmd = &cobra.Command{
 	Short: "Use eBPF to speed up your Service Mesh like crossing an Einstein-Rosen Bridge.",
 	Long:  `Use eBPF to speed up your Service Mesh like crossing an Einstein-Rosen Bridge.`,
 	RunE: func(cmd *cobra.Command, args []string) error {
-		s := cniserver.NewServer("/var/run/merbridge-cni.sock", "/sys/fs/bpf")
+		s := cniserver.NewServer("/host/var/run/merbridge-cni.sock", "/sys/fs/bpf")
```
Comment (kebe7jun) [inline app/cmd/root.go:41]: It's a good idea to make the `/host` directory configurable so that we can also debug it or use it for e2e again.

**Comment by kebe7jun** [comment]
At this stage, we may consider adding a startup parameter(like `--cni-mode` ?) to ensure that `install-cni` is not turned on by default, as this may not have been rigorously tested.
We will enable it by default when it is stable.

**Comment by kebe7jun** [comment]
> At this stage, we may consider adding a startup parameter(like `--cni-mode` ?) to ensure that `install-cni` is not turned on by default, as this may not have been rigorously tested. We will enable it by default when it is stable.

If this pr merged, the merbridge cni will be installed to all of our users, so, we should take care.

File: app/cmd/root.go (hunk 4 above)
Comment (dddddai) [reply app/cmd/root.go:41]: I think this is in the helm part, but I'm not familiar with helm yet, we can do this in a follow-up

**Comment by Xunzhuo** [comment]
> At this stage, we may consider adding a startup parameter(like `--cni-mode` ?) to ensure that `install-cni` is not turned on by default, as this may not have been rigorously tested.
> 
> We will enable it by default when it is stable.

+1

**Review by kebe7jun — APPROVED** [review]

**Pull request merged**